juniper/src/types/async_await.rs

```diff
@@ -306,7 +306,11 @@ where
                 );
 
                 let concrete_type_name = instance.concrete_type_name(sub_exec.context(), info);
-                if fragment.type_condition.item == concrete_type_name {
+                let type_name = instance.type_name(info);
+
+                if fragment.type_condition.item == concrete_type_name
+                    || Some(fragment.type_condition.item) == type_name
+                {
```

Comment:
@ArsileLuci you've considered async resolving only. We need to consider sync resolving too.